### DISCUSSION TITLE
Referencing an External Workflow template in Application

### DISCUSSION THREAD

**Discussion by testinfected** [kubevela/kubevela#6401] [Q&A] [answered]
Hi all,

I'd like to know if I can reference an external workflow template in an Application. That feature is mentionned in WorkflowRuns, and seem supported in Applications as well. Indeed the Application schema defines a `Ref` field in workflow as an alternative to `Steps`.

I've not been able to make it work though, it's like that reference is ignored and Vela uses the default workflow as if no workflow at all was specified.

Any idea what I'm doing wrong?

**Comment by FogDong** [comment, marked as answer]
You can checkout this doc and use dry-run to see if the application you rendered is what you want: https://kubevela.io/docs/tutorials/dry-run/#dry-run-with-policy-and-workflow-via-cli

**Reply by testinfected** [reply]
Not sure what I dig wrong in the first place, but yes it works following that example. I will build from that. Thanks
